durabletask-azurefunctions/durabletask/azurefunctions/decorators/durable_app.py

```diff
@@ -0,0 +1,266 @@
+
+from durabletask.internal.orchestrator_service_pb2 import OrchestratorRequest, OrchestratorResponse
+from .metadata import OrchestrationTrigger, ActivityTrigger, EntityTrigger, \
+    DurableClient
+from typing import Callable, Optional
+from typing import Union
+from azure.functions import FunctionRegister, TriggerApi, BindingApi, AuthLevel
+
+# TODO: Use __init__.py to optimize imports
+from durabletask.azurefunctions.client import DurableFunctionsClient
+from durabletask.azurefunctions.worker import DurableFunctionsWorker
+from durabletask.azurefunctions.internal.azurefunctions_null_stub import AzureFunctionsNullStub
+
+
+class Blueprint(TriggerApi, BindingApi):
+    """Durable Functions (DF) Blueprint container.
+
+    It allows functions to be declared via trigger and binding decorators,
+    but does not automatically index/register these functions.
+
+    To register these functions, utilize the `register_functions` method from any
+    :class:`FunctionRegister` subclass, such as `DFApp`.
+    """
+
+    def __init__(self,
+                 http_auth_level: Union[AuthLevel, str] = AuthLevel.FUNCTION):
+        """Instantiate a Durable Functions app with which to register Functions.
+
+        Parameters
+        ----------
+        http_auth_level: Union[AuthLevel, str]
+            Authorization level required for Function invocation.
+            Defaults to AuthLevel.Function.
+
+        Returns
+        -------
+        DFApp
+            New instance of a Durable Functions app
+        """
+        super().__init__(auth_level=http_auth_level)
+
+    def _configure_orchestrator_callable(self, wrap) -> Callable:
+        """Obtain decorator to construct an Orchestrator class from a user-defined Function.
+
+        Parameters
+        ----------
+        wrap: Callable
+            The next decorator to be applied.
+
+        Returns
+        -------
+        Callable
+            The function to construct an Orchestrator class from the user-defined Function,
+            wrapped by the next decorator in the sequence.
+        """
+        def decorator(orchestrator_func):
+            # Construct an orchestrator based on the end-user code
+
+            # TODO: Move this logic somewhere better
+            def handle(context) -> str:
+                context_body = getattr(context, "body", None)
+                if context_body is None:
+                    context_body = context
+                orchestration_context = context_body
+                request = OrchestratorRequest()
+                request.ParseFromString(base64.b64decode(orchestration_context))
+                stub = AzureFunctionsNullStub()
+                worker = DurableFunctionsWorker()
+                response: Optional[OrchestratorResponse] = None
+
+                def stub_complete(stub_response):
+                    nonlocal response
+                    response = stub_response
+                stub.CompleteOrchestratorTask = stub_complete
```

Comment:
All of this is probably optimizable - do we really need to create a new stub and worker for each call? Can they be saved? Will look into this more at some point